src/platform/unix/linux/mod.rs

```diff
@@ -26,18 +26,64 @@ use std::thread;
 
 const MAX_FDS_IN_CMSG: u32 = 64;
 
+const SCM_RIGHTS: c_int = 0x01;
+
+// The value Linux returns for SO_SNDBUF
+// is not the size we are actually allowed to use...
+// Empirically, we have to deduct 32 bytes from that.
+const RESERVED_SIZE: usize = 32;
+
+#[cfg(target_os="android")]
+const TEMP_FILE_TEMPLATE: &'static str = "/sdcard/servo/ipc-channel-shared-memory.XXXXXX";
+
+#[cfg(not(target_os="android"))]
+const TEMP_FILE_TEMPLATE: &'static str = "/tmp/ipc-channel-shared-memory.XXXXXX";
+
+
+#[cfg(target_os="linux")]
+type IovLen = usize;
+#[cfg(target_os="linux")]
+type SockLen = usize;
```

Comment:
That's what I was hoping. However, [socklen_t:linux = u32](https://doc.rust-lang.org/libc/x86_64-unknown-linux-gnu/libc/type.socklen_t.html) but [msghdr.msg_controllen = size_t = usize](https://doc.rust-lang.org/libc/x86_64-unknown-linux-gnu/libc/struct.msghdr.html). This works on FreeBSD though, and was what I originally did. Perhaps this is another change that should be made to `libc`